src/recommendpolicy/recommendPolicy.go

```diff
@@ -223,7 +205,55 @@ func uniqueNsDeploy(deployName, deployNamespace string) *types.Deployment {
 	return &deploy
 }
 
-func GetHardenPolicy(deployments *v1.DeploymentList, replicaSets *v1.ReplicaSetList, statefulSets *v1.StatefulSetList, daemonSets *v1.DaemonSetList, nsNotFilter []string) []types.KnoxSystemPolicy {
+func GetAdmissionControllerPolicy(deployments, replicaSets, statefulSets, daemonSets []types.Deployment) []v1.Policy {
+
+	var policies []v1.Policy
+
+	nsNotFilterAdmissionControllerPolicy := cfg.CurrentCfg.ConfigAdmissionControllerPolicy.NsNotFilter
+	nsFilterAdmissionControllerPolicy := cfg.CurrentCfg.ConfigAdmissionControllerPolicy.NsFilter
+	recommendAdmissionControllerPolicy := cfg.GetCfgRecommendAdmissionControllerPolicy()
+
+	for _, d := range deployments {
+
+		labelMap := labelArrayToLabelMap(strings.Split(d.Labels, ","))
+
+		if recommendAdmissionControllerPolicy &&
+			isNamespaceAllowed(d.Namespace, nsNotFilterAdmissionControllerPolicy, nsFilterAdmissionControllerPolicy) {
+			policies = append(policies, generateAdmissionControllerPolicy(d.Name, d.Namespace, labelMap)...)
+		}
+	}
+	for _, d := range replicaSets {
+
+		labelMap := labelArrayToLabelMap(strings.Split(d.Labels, ","))
+
+		if recommendAdmissionControllerPolicy &&
+			isNamespaceAllowed(d.Namespace, nsNotFilterAdmissionControllerPolicy, nsFilterAdmissionControllerPolicy) {
+			policies = append(policies, generateAdmissionControllerPolicy(d.Name, d.Namespace, labelMap)...)
+		}
+	}
+	for _, d := range statefulSets {
+
+		labelMap := labelArrayToLabelMap(strings.Split(d.Labels, ","))
+
+		if recommendAdmissionControllerPolicy &&
+			isNamespaceAllowed(d.Namespace, nsNotFilterAdmissionControllerPolicy, nsFilterAdmissionControllerPolicy) {
+			policies = append(policies, generateAdmissionControllerPolicy(d.Name, d.Namespace, labelMap)...)
+		}
+	}
+	for _, d := range daemonSets {
+
+		labelMap := labelArrayToLabelMap(strings.Split(d.Labels, ","))
+
+		if recommendAdmissionControllerPolicy &&
+			isNamespaceAllowed(d.Namespace, nsNotFilterAdmissionControllerPolicy, nsFilterAdmissionControllerPolicy) {
+			policies = append(policies, generateAdmissionControllerPolicy(d.Name, d.Namespace, labelMap)...)
+		}
+	}
```

Comment:
This won't work for all types of pod controllers as we generate some fields in Kyverno policy according to the type of controller. 

https://github.com/accuknox/discovery-engine/blob/1746c3ec0dcc8d514ac826bdc0e6a2b9b4c0a0b1/src/recommendpolicy/helperFunctions.go#L148-L151

Other generated fields would work similarly for all pod controllers except `CronJob`.

---

So you would need to have the name of the kind passed while generating Kyverno policy so it gets updated accordingly. The merger of https://github.com/accuknox/discovery-engine/pull/734 would lead to some conflicts as we are shifting to kyverno policy interface.